contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -339,13 +267,29 @@ contract RibbonTreasuryVault is
             "Invalid management fee"
         );
 
-        // We are dividing annualized management fee by num weeks in a year
-        uint256 tmpManagementFee =
-            newManagementFee.mul(Vault.FEE_MULTIPLIER).div(WEEKS_PER_YEAR);
+        managementFee = _perRoundManagementFee(newManagementFee);
 
         emit ManagementFeeSet(managementFee, newManagementFee);
+    }
 
-        managementFee = tmpManagementFee;
+    /**
+     * @notice Internal function to set the management fee for the vault
+     * @param managementFee is the management fee (6 decimals). ex: 2 * 10 ** 6 = 2
+     * @return perRoundManagementFee is the management divided by the number of rounds per year
+     */
+    function _perRoundManagementFee(uint256 managementFee)
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 _period = period;
+        uint256 feeDivider =
+            _period % 30 == 0
+                ? Vault.FEE_MULTIPLIER.mul(12 / (_period / 30))
```

Comment:
Why do we need to multiply by `Vault.FEE_MULTIPLIER` here? I see we are already multiplying it on line 292

Reply:
This is to follow the `WEEKS_PER_YEAR` convention.

```Number of weeks per year = 52.142857 weeks * FEE_MULTIPLIER = 52142857```

For the monthly, quarterly and biannually, we need to do the same. So if we have monthly, the management fee will be divided by 12000000.